packages/node-resolve/test/test.js

```diff
@@ -124,6 +141,25 @@ test('supports JS extensions in TS when referring to TS imports', async (t) => {
   t.is(module.exports, 'It works!');
 });
 
+test('supports JS extensions in TS actually importing JS with export map', async (t) => {
```

Comment:
What about the following cases:

* `.js` file importing `.ts` file?
* `.ts` file importing `.ts` file?
* `.ts` file importing another `.ts` file using the `.js` extension?

Reply:
I think these should all be fine because they will all be imports with the extension included?

I think
>.ts file importing another .ts file using the .js extension?

is covered with "supports JS extensions in TS when referring to TS imports"?

Reply:
There is an ambiguity with the `.ts` file loading a `.js` file case - because it could be a `.js` or a `.ts`, and then the question is which to give precedence. Is there an existing test covering this case? If not it would be good to add one to ensure this is no regression here.

Reply:
Sure happy to add more tests. Still not following exactly what you mean though. Do you mean a .ts file importing `file.js`, when there is a `file.js` and a `file.ts`, and checking the `file.js` is actually the one that's loaded?

Reply:
Yes that's the case, where the `file.ts` is loaded when the `file.js` does not exist.

Reply:
There's
- "supports JS extensions in TS actually importing JS with export map"
- "supports JS extensions in TS when referring to TS imports"

I also just updated the test I added to have an import with the same name to ensure it prefers the one in the import and doesn't always switch to ".ts"

Reply:
Great, thanks for confirming this.